pyccel/ast/builtins.py

```diff
@@ -260,7 +260,9 @@ class PythonFloat(PyccelAstNode):
     _attribute_nodes = ('_arg',)
 
     def __new__(cls, arg):
-        if isinstance(arg, LiteralFloat):
+        if isinstance(arg.dtype, NativeBool):
+            return PythonInt(arg)
+        elif isinstance(arg, LiteralFloat):
             return LiteralFloat(arg, precision = cls._precision)
```

Comment:
This may not be totally relevant to this PR, but it's quite close. Given that a `LiteralFloat` object is essentially immutable, can't we simply return the argument in this case? That is
```python
elif isinstance(arg, LiteralFloat):
    return arg
```

Reply:
You need be sure that you have the right precision. So this would be:
```
elif isinstance(arg, LiteralFloat) and arg.precision == cls._precision:
    return arg
elif isinstance(arg, LiteralFloat):
    return LiteralFloat(arg, precision = cls._precision)
```

Reply:
But the precision is in the class and it is fixed, or am I missing something?

Reply:
The precision of the result is fixed, but that doesn't mean that it matches the precision of the argument.
Imagine if you do:
```
x = np.float32(3)
y = float(x)
```
With your code y will still be a `float32` instead of a `float64`

Reply:
I thought that `x` would be a `Variable` in your example, not a `LiteralFloat`.

Reply:
Ah you are right, sorry the example would have to be:
```
y = float(np.float32(3))
```

Reply:
OK, I see the problem now. @mostafaMamoni you can write this:
```python
elif isinstance(arg, LiteralFloat):
    prec = cls._precision
    return arg if arg.precision == prec else LiteralFloat(arg, precision = prec)
```

Reply:
By the way, why do we pass `arg` to `LiteralFloat` instead of `arg.p`?

Reply:
`arg.p` is a sympy relic that only ever worked for integers and is removed in PR #764.
We could use `arg.python_value` though as this is what `LiteralFloat` will do in the `__init__` method anyway

Reply:
>`arg.p` is a sympy relic that only ever worked for integers and is removed in PR #764.

Nice!

I don't want to cause confusion. Whatever seems more appropriate for this PR.

Reply:
do you mean `arg.python_value` ?. you can find `p` attribute in `LiteralInteger` only.

Reply:
OK, use `python_value` then:
```python
elif isinstance(arg, LiteralFloat):
    ap = arg.precision
    cp = cls._precision
    return arg if ap == cp else LiteralFloat(arg.python_value, precision = cp)
```

Reply:
@EmilyBourne Should we leave this to PR #764 instead?

Reply:
Yes i think it makes more sense to have this discussion there